subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/maven/MavenHttpRepoResolveIntegrationTest.groovy

```diff
@@ -431,4 +431,43 @@ task listJars {
 
         errorOutput.contains('Response 304: Not Modified has no content!')
     }
+
+    def "does not query Maven repository for modules without a group, name or version"() {
+        given:
+        buildFile << """
+            repositories {
+                maven {
+                    url '${mavenHttpRepo.uri}'
+                }
+            }
+            configurations { compile }
+            dependencies { 
+                compile ':name1:1.0' 
+                compile ':name2:[1.0, 2.0]' 
+                compile ':name3:1.0-SNAPSHOT'
+                compile 'group1::1.0' 
+                compile 'group2::[1.0, 2.0]' 
+                compile 'group3::1.0-SNAPSHOT'
+                compile 'group:name'
+            }
+            task resolve {
+                doLast {
+                    assert configurations.compile.resolve()
+                }
+            }
+        """
+
+        when:
+        fails 'resolve'
+
```

Comment:
It would be nice if there was a way to explicitly declare `expectNoHttpInteractions()` here.

Reply:
We can add such a method, though it's just a no-op. The fixture will complain about unexpected interactions.